repo/encryption/aes256_gcm_hmac_sha256_encryptor.go

```diff
@@ -0,0 +1,63 @@
+package encryption
+
+import (
+	"crypto/aes"
+	"crypto/cipher"
+	"crypto/hmac"
+	"crypto/sha256"
+
+	"github.com/pkg/errors"
+)
+
+var zeroAES256GCMNonce = make([]byte, 12)
+
+type aes256GCMHmacSha256 struct {
+	keyDerivationSecret []byte
+}
+
+// aeadForContent returns cipher.AEAD using key derived from a given contentID.
+func (e aes256GCMHmacSha256) aeadForContent(contentID []byte) (cipher.AEAD, error) {
+	h := hmac.New(sha256.New, e.keyDerivationSecret)
```

Comment:
Not sure where the right place would be, but it may make sense to enforce a minimum length for `keyDerivationSecret` to prevent incorrect use.
Notice that:
* since this is not the key directly passed to the cipher below, the cipher does not validate that `keyDerivationSecret` has a minimum length
* the `init()` below generated as 32-byte key, but there are no checks to prevent accidentally creating a short `keyDerivationSecret`

Reply:
it just so happens that deriveKey() always takes 32 bytes today (that's the reason for // nolint:unparam). I added error condition to make sure it fails for attempts to derive shorter keys.